executor/src/witgen/affine_expression.rs

```diff
@@ -235,27 +235,31 @@ where
         }
 
         // Check if they are mutually exclusive and compute assignments.
-        let mut covered_bits: AbstractNumberType = 0;
+        let mut covered_bits: <FieldElement as FieldElementTrait>::Integer = 0u32.into();
         let mut assignments = EvalValue::complete(vec![]);
         let mut offset = (-self.offset).to_integer();
         for (i, coeff, constraint) in parts {
             let constraint = constraint.clone().unwrap();
             let mask = constraint.mask();
-            if mask & covered_bits != 0 {
+            if *mask & covered_bits != 0u32.into() {
                 return Ok(EvalValue::incomplete(
                     IncompleteCause::OverlappingBitConstraints,
                 ));
             } else {
-                covered_bits |= mask;
+                covered_bits |= *mask;
             }
             assignments.combine(EvalValue::complete(vec![(
                 i,
-                Constraint::Assignment(((offset & mask) / coeff.to_integer()).into()),
+                Constraint::Assignment(
+                    ((offset & *mask).into_biguint() / coeff.to_arbitrary_integer())
+                        .try_into()
+                        .unwrap(),
+                ),
             )]));
-            offset &= !mask;
+            offset &= !*mask
```

Comment:
Why is this returned now?

Reply:
Interesting, `&=` returns nothing so these are equivalent I think.

Reply:
Then maybe let's keep `;`?